editor/editor_properties_array_dict.cpp

```diff
@@ -480,7 +485,28 @@ bool EditorPropertyArray::_is_drop_valid(const Dictionary &p_drag_data) const {
 		return true;
 	}
 
-	return false;
+	if (p_drag_data["type"] != "nodes") {
+		return false;
+	}
+
+	if (allowed_type.is_empty()) {
```

Comment:
This condition looks like something that should be at the beginning.
Also what is this supposed to do? If it's to allow any valid type to plain Array, it doesn't work.